status/raw/status-app-protocols.md

```diff
@@ -0,0 +1,372 @@
+
+**Large messages**
+
+Status clients (full or light) SHOULD use store queries (rather than subscriptions) to retrieve large messages relevant to that client.
+See [Large messages](#large-messages).
+
+### Providing services
+
+Status clients MAY provide service-side protocols to other clients.
+
+Full clients SHOULD mount
+the filter service protocol (see [12/WAKU2-FILTER](../../waku/standards/core/12/filter.md))
+and lightpush service protocol (see [19/WAKU2-LIGHTPUSH](../../waku/standards/core/19/lightpush.md))
+in order to provide light subscription and publishing services to other clients 
+for each pubsub topic to which they have a relay subscription.
+
+Status clients MAY mount the store query protocol as service node (see [WAKU2-STORE](https://github.com/waku-org/specs/blob/8fea97c36c7bbdb8ddc284fa32aee8d00a2b4467/standards/core/store.md))
+to store historical messages and
+provide store services to other clients
+for each pubsub topic to which they have a relay subscription
+
+### Self-addressed messages
+
+Messages with a _local_ functional scope (see [Functional scope](#functional-scope)),
+also known as _self-addressed_ messages,
+MUST be published to a distinct pubsub topic or a distinct _set_ of pubsub topics
+used exclusively for messages with local scope (see [Pubsub topics and sharding](#pubsub-topics-and-sharding)).
+Status clients (full or light) MUST use lightpush protocol to publish self-addressed messages (see [Publishing](#publishing)).
+Status clients (full or light) MUST NOT subscribe to topics for messages with self-addressed scopes (see [Subscribing](#subscribing)).
+Status clients (full or light) SHOULD use store queries (rather than subscriptions) to retrieve self-addressed messages relevant to that client (see [Retrieving historical messages](#retrieving-historical-messages)).
+
+### Large messages
+
+The application MAY define separate pubsub topics for large messages.
+These pubsub topics for large messages MAY be distinct for each functional scope (see [Pubsub topics and sharding](#pubsub-topics-and-sharding)).
+Status clients (full or light) SHOULD use lightpush protocols to publish to pubsub topics set aside for large messages (see [Publishing](#publishing)).
+Status clients (full or light) SHOULD NOT subscribe to topics set aside for large messages (see [Subscribing](#subscribing)).
+Status clients (full or light) SHOULD use store queries (rather than subscriptions) to retrieve large messages relevant to that client (see [Retrieving historical messages](#retrieving-historical-messages)).
+
+**Chunking**
+
+The Status application MAY use a chunking mechanism to break down large payloads
+into smaller segments for individual Waku transport.
+The definition of a large message is up to the application.
+However, the maximum size for a [14/WAKU2-MESSAGE](../../waku/standards/core/14/message.md) payload is 150KB.
+Status application payloads that exceed this size MUST be chunked into smaller pieces
```

Comment:
isn't this statement contradicting in a way. 
it states that large payloads must be chunked and also considered for large messages which mean they use a dedicated large-messages pubsubtopic.

I am wondering if pubsubtopics/shards are going to be dedicated for large messages do we need to set the 150KB limit or increase it since clients anyways are not going to subscribe to these topics.

Reply:
> it states that large payloads must be chunked and also considered for large messages which mean they use a dedicated large-messages pubsubtopic.

I expect large messages to be chunked, in more chunks.

> I am wondering if pubsubtopics/shards are going to be dedicated for large messages do we need to set the 150KB limit or increase it since clients anyways are not going to subscribe to these topics.

Fair q.

Reply:
> I am wondering if pubsubtopics/shards are going to be dedicated for large messages do we need to set the 150KB limit or increase it since clients anyways are not going to subscribe to these topics.

Indeed a fair question to consider in future, but I don't think needs to be defined right now. The point is that the total data transfer for large messages (whether in chunks or not) needs a designed mechanism to (1) prevent them from being broadcast to everyone (2) allow users to download ad-hoc.